File: java/program/Dragon.java
public class Dragon extends Reptile {
    private boolean canFly;

    public Dragon(String name, double length, String habitat, String diet, boolean canFly) {
        super(name, length, habitat, diet);
        this.canFly = canFly;
    }

    @Override
    public void display() {
        System.out.println("Dragon: " + name +
                           ", Can Fly: " + (canFly ? "Yes" : "No") +
                           ", Habitat: " + habitat);
    }
}

File: cpp/Program/Dragon.cpp
#include "Dragon.h"

Dragon::Dragon(string n, double l, string h, string d, bool f)
    : Reptile(n, l, h, d), canFly(f) {}

void Dragon::display() {
    cout << "Dragon: " << name
         << ", Can Fly: " << (canFly ? "Yes" : "No")
         << ", Habitat: " << habitat << endl;
}
